source/scripts/ContentScript/inject/stargazerProvider.txt.ts

```diff
@@ -34,18 +34,22 @@ const SUPPORTED_CHAINS = {
   }
 };
 
+class RPCError {
+  constructor(message, code) {
+    this.message = message;
+    this.name = 'Stargazer Wallet Error';
+    this.code = code;
+  }
+}
+
 const ERRORS = {
   USER_REJECTED: (message = 'User Rejected Request') => {
-    const err = new Error(message);
-    err.code = 4001;
-    return err;
+    return new RPCError(message, 4001);
   },
   INVALID_METHOD: (message = 'Unsupported Method') => {
-    const err = new Error(message);
-    err.code = 4200;
-    return err;
+    return new RPCError(message, 4200);
```

Comment:
instead of using Error object, just to create own class RPCError which sends back object with keys message, name, and code.